geopandas/plotting.py

```diff
@@ -238,52 +412,82 @@ def plot_dataframe(s, column=None, cmap=None, color=None, linewidth=1.0,
         return plot_series(s.geometry, cmap=cmap, color=color,
                            ax=ax, linewidth=linewidth, figsize=figsize,
                            **color_kwds)
+
+    if s[column].dtype is np.dtype('O'):
+        categorical = True
+
+    # Define `values` as a Series
+    if categorical:
+        if cmap is None:
+            cmap = 'Set1'
+        categories = list(set(s[column].values))
+        categories.sort()
+        valuemap = dict([(k, v) for (v, k) in enumerate(categories)])
+        values = pd.Series([valuemap[k] for k in s[column]])
     else:
-        if s[column].dtype is np.dtype('O'):
-            categorical = True
+        values = s[column]
+    if scheme is not None:
+        binning = __pysal_choro(values, scheme, k=k)
+        values = pd.Series(binning.yb)
+        # set categorical to True for creating the legend
+        categorical = True
+        binedges = [binning.yb.min()] + binning.bins.tolist()
+        categories = ['{0:.2f} - {1:.2f}'.format(binedges[i], binedges[i+1])
+                      for i in range(len(binedges)-1)]
+    if ax is None:
+        fig, ax = plt.subplots(figsize=figsize)
+        ax.set_aspect('equal')
+
+    mn = values.min() if vmin is None else vmin
+    mx = values.max() if vmax is None else vmax
+
+    # plot all Polygons and all MultiPolygon components in the same collection
+    poly_idx = (s.geometry.type == 'Polygon') | (s.geometry.type == 'MultiPolygon')
+    polys = s.geometry[poly_idx]
+    if not polys.empty:
+        # Plot the fill with default or user-specified alpha, but do not draw
+        # outlines.
+        plot_polygon_collection(ax, polys, values[poly_idx], True,
+                                vmin=mn, vmax=mx, cmap=cmap,
+                                linewidth=0, **color_kwds)
+        # Draw the edges, fully opaque, but no facecolor.
+        edges_kwds = color_kwds.copy()
+        edges_kwds['alpha'] = 1
+        edges_kwds['facecolor'] = 'none'
+        # Setting plot_values=False would cause the array values' colors to
+        # override edgecolor. By setting color instead, matplotlib will respect
+        # edgecolor if set.
+        plot_polygon_collection(ax, polys, ['black'] * len(polys), False,
```

Comment:
@IamJeffG, this second `plot_polygon_collection` is unnecessarily called in situations where linewidth=0 or alpha=1. In my test data (that I can't share :D), plotting a GeoDataFrame takes about twice as long when executing this second call. I'll open a PR against your fork so you can propagate it over here if you want.

I hope life after TCC is treating you well!